tests/unit/test_utils.py

```diff
@@ -146,6 +146,8 @@ def test_open_guess_encoding():
         ("git+https://github.com/jupyterhub/repo2docker", False),
         ("numpy", False),
         ("# -e .", False),
+        ("--pre", False),
+        ("--pre pandas", False),
```

Comment:
Can you add a test for:

    "--unrecognized"

and make sure it doesn't error (I think it will)

Reply:
also `"-e"` with nothing after it